uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Wrapper/LeaveRequestFieldsVisibility.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_Service_LeaveRequestRights as LeaveRequestRightsService;
+use CRM_HRLeaveAndAbsences_Service_LeaveManager as LeaveManagerService;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestFieldsVisibility
+ */
+class CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestFieldsVisibility implements API_Wrapper {
+
+  /**
+   * @var array
+   *   An array of API actions for the LeaveRequest Entity with properties for
+   *   each action that determine which handler should handle which API and also
+   *   if the contact_id should be among the fields to be returned for the API
+   *   for the respective handler to work properly. Since the Identifier field used
+   *   by some of the handler to process the field values to be hidden is the contact_id,
+   *   There will be unreliable results if this field is absent in the results set.
+   */
+  protected $apiActions = [
+    'getfull' => [
+      'handler' => 'CRM_HRLeaveAndAbsences_API_Handler_GenericLeaveFieldPermissions',
+      'request_requires_contact_id' => TRUE
+    ],
+    'get' => [
+      'handler' => 'CRM_HRLeaveAndAbsences_API_Handler_GenericLeaveFieldPermissions',
+      'request_requires_contact_id' => TRUE
+    ],
+    'getbreakdown' => [
+      'handler' => 'CRM_HRLeaveAndAbsences_API_Handler_GetBreakDownFieldPermissions'
+    ],
+  ];
+
+  /**
+   * @param $apiRequest
+   * @return mixed
+   */
+  public function fromApiInput($apiRequest) {
+    if ($this->canHandleApiInput($apiRequest)) {
```

Comment:
Please explain on the dockblock what you're doing with the contact_id here and why